### PR TITLE
[serve] Update quickstart & getting started for explicit `Application` concept

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Docs follow up to: https://github.com/ray-project/ray/pull/34627

First PR in a series to standardize around the `Application` concept.

- Updates quickstarts & getting started w/ relevant verbiage.
- De-emphasizes deployment graph API (that should be introduced in its own section).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/getting_started.md [hunk 1]
```diff
@@ -178,9 +174,10 @@ $ python model_client.py
 Bonjour monde!
 ```
 
-## Composing Machine Learning Models with Deployment Graphs
+## Composing Multiple Models
 
-Ray Serve's Deployment Graph API allows us to compose multiple machine learning models together into a single Ray Serve application. We can use parameters like `num_replicas`, `num_cpus`, and `num_gpus` to independently configure and scale each deployment in the graph.
+Ray Serve allows you to compose multiple deployments into a single Ray Serve application. This makes it easy to combine multiple machine learning models along with business logic to serve a single request.
+We can use parameters like `num_replicas`, `num_cpus`, and `num_gpus` to independently configure and scale each deployment in the application.
```
Comment (akshay-anyscale) [inline doc/source/serve/getting_started.md:180]: and autoscaling config/?

**Review by akshay-anyscale — APPROVED** [review]
approved with small nit

File: doc/source/serve/getting_started.md (hunk 1 above)
Comment (edoakes) [reply doc/source/serve/getting_started.md:180]: good point

Comment (edoakes) [reply doc/source/serve/getting_started.md]: done

**Review by akshay-anyscale — APPROVED** [review]

**Review by sihanwang41 — APPROVED** [review]

**Pull request merged**